src/raft.c

```diff
@@ -434,17 +461,28 @@ RedisModuleCallReply *RaftExecuteCommandArray(RedisRaftCtx *rr,
          * When we have an ACL, we will have a user set on the context, so need "C"
          */
         char *resp_call_fmt;
-        if (cmds->cmd_flags & CMD_SPEC_MULTI) {
+        if (cmds->cmd_flags & CMD_SPEC_MULTI || client_session) {
             /* don't block inside a MULTI */
             resp_call_fmt = cmds->acl ? "CE0v" : "E0v";
         } else {
             resp_call_fmt = cmds->acl ? "KCE0v" : "KE0v";
         }
 
         enterRedisModuleCall();
-        RedisModule_SetContextUser(ctx, user);
+        if (client_session) {
+            RedisModule_SetClientUser(client_session->client, user);
+            RedisModule_SetContextClient(ctx, client_session->client);
+        } else {
+            RedisModule_SetContextUser(ctx, user);
+        }
         reply = RedisModule_Call(ctx, cmd, resp_call_fmt, &c->argv[1], c->argc - 1);
-        RedisModule_SetContextUser(ctx, NULL);
+        if (client_session) {
+            RedisModule_SetClientUser(client_session->client, NULL);
+            RedisModule_SetContextClient(ctx, NULL);
+        } else {
+            RedisModule_SetContextUser(ctx, NULL);
+        }
+        RedisModule_SetContextClient(ctx, NULL);
```

Comment:
why do we need to do it twice?

Reply:
what twice?  in one case we set the ctx's user to null, in the other case we set the ctx's client to null.

Reply:
in `client_session == True`  case, we already set the ctx's client => if we set the ctx's client to null anyway, no need to set it first at `client_session` case

Reply:
yes, we could check it / wrap it in more if/else conditions, but setting it to null always afterwards, even if set up front, seems cleaner.